src/components/common/PinButton.js

```diff
@@ -0,0 +1,82 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import IconButton from '@material-ui/core/IconButton';
+import { Map } from 'immutable';
+import PushPinIcon from '@material-ui/icons/PushPin';
+import PushPinOutlinedIcon from '@material-ui/icons/PushPinOutlined' 
+import { useTranslation } from 'react-i18next';
+import Tooltip from '@material-ui/core/Tooltip';
+import { MUTATION_KEYS } from '@graasp/query-client';
+import { useMutation, hooks } from '../../config/queryClient';
+import { FAVORITE_ITEM_BUTTON_CLASS } from '../../config/selectors';
+import { getItemPinnedTag, isItemPinned } from '../../utils/itemTag';
+import { getItemPinnedTagFromItem } from '../../utils/itemExtra'
+
+const { useTags, useItemTags } = hooks;
+
+const PinButton = ({ item, member }) => {
+  const { t } = useTranslation();
+  const { data: tags } = useTags();
+
+  const addPinned = useMutation(MUTATION_KEYS.POST_ITEM_TAG);
```

Comment:
An elegant trick is to have `{ mutate: addPin }` so that you can write `addPin(payload)`